### PR TITLE
[HOLD] SocksProxy: unit-tests, functional refactors for testing, and inlining refactors

### PR DESCRIPTION
**Pull request by coneiric**
---
**By submitting this pull-request, I confirm the following:**

- I have read and understood the developer guide in [kovri-docs](https://github.com/monero-project/kovri-docs).
- I have checked that another pull-request for this purpose does not exist.
- I have considered and confirmed that this submission will be valuable to others.
- I accept that this submission may not be used and that this pull-request may be closed by the will of the maintainer.
- I give this submission freely under the BSD 3-clause license.
---

### CODE REVIEW

File: src/client/proxy/socks.h [hunk 1]
```diff
@@ -334,9 +340,15 @@ class SOCKSHandler
       SOCKSServer* parent,
       std::shared_ptr<boost::asio::ip::tcp::socket> socket);
 
-  ~SOCKSHandler();
+  ~SOCKSHandler()
+  {
+    Terminate();
```
Comment (anonimal) [inline src/client/proxy/socks.h]: No, move back. These are not small functions. This is noted in https://google.github.io/styleguide/cppguide.html#Inline_Functions.

File: src/client/proxy/socks.h [hunk 2]
```diff
@@ -334,9 +340,15 @@ class SOCKSHandler
...
+  }
 
-  void Handle();
+  void Handle()
+  {
+    AsyncSocketRead();
```
Comment (anonimal) [inline src/client/proxy/socks.h]: No, move back. These are not small functions. This is noted in https://google.github.io/styleguide/cppguide.html#Inline_Functions.

File: src/client/proxy/socks.h (hunk 1 above)
Comment (coneiric) [reply src/client/proxy/socks.h]: Alright, I misunderstood again. Didn't realize that the line count applied through the inner function call.

File: src/client/proxy/socks.h (hunk 2 above)
Comment (coneiric) [reply src/client/proxy/socks.h]: Alright, I misunderstood again. Didn't realize that the line count applied through the inner function call.

File: src/client/proxy/socks.h (hunk 1 above)
Comment (anonimal) [reply src/client/proxy/socks.h]: Always, because that is the code path.

File: src/client/proxy/socks.h (hunk 2 above)
Comment (anonimal) [reply src/client/proxy/socks.h]: Always, because that is the code path.

**Pull request closed without merging**